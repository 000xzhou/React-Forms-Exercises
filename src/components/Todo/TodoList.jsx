function TodoList() {
  return "<BoxList />";
}

export default TodoList;
