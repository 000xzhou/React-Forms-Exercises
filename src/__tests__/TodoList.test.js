import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "../components/Todo/TodoList";

it("should render without crashing", () => {
  render(<TodoList />);
});

it("should match snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add new todo", () => {
  const { getByLabelText, getByText } = render(<TodoList />);
  // get inputs
  const todoInput = getByLabelText("Todo:");
  const addButton = getByText("Add");

  // add input and submit
  fireEvent.change(todoInput, { target: { value: "100" } });
  fireEvent.click(addButton);
});
