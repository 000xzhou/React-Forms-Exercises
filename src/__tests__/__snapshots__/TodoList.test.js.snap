// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todoInput"
    >
      Todo:
    </label>
    <input
      id="todoInput"
      name="todoInput"
      placeholder="Eat dinner"
      type="text"
      value=""
    />
    <button>
      Add
    </button>
  </form>
</DocumentFragment>
`;
